### ISSUE TITLE
Add support for custom authentication methods towards maven repository

### ISSUE THREAD

**Issue by rheinertleon** [jeka-dev/jeka#233]
I currently try to use Jeka to dynamically load plugin jars hosted in a private maven repository.
Specifically, they are hosted in a private Gitlab-Instance, which has [a specific way of authentication](https://docs.gitlab.com/ee/user/packages/maven_repository/#edit-the-settingsxml).

If I understand correctly, it requires special HTTP-Header to be set to a generated token instead of basic auth (user+password).
E.g., my maven `setting.xml` includes this in the `servers`-section:

```xml
<server>
	<id>gitlab-maven</id>
	<configuration>
		<httpHeaders>
			<property>
				<name>Private-Token</name>
				<value>my-token-here</value>
			</property>
		</httpHeaders>
	</configuration>
</server>
```
Instead of `Private-Token`, it could also be `Deploy-Token` or `Job-Token`.

I did not find any way of configuring such a logic for Jeka.
When I tried to resolve a private artifact, I would expect an error concerning authentication, but instead I received the following error.

`Exception in thread "main" java.lang.IllegalStateException: Errors with dependencies : [com.exampl:artifact:1.0.0 -> not found] `

**Comment by djeang** [comment]
You are right, there is no support for this  it but I'll try to add it asap. One solution might consist in setting a property like
```
jeka.repos.download.headers.Private-Token=my-token-here
```
What do you think of ?

**Comment by rheinertleon** [comment]
This would definitely be sufficient for me.

I wondered how this can also be possible through the API, to keep this nice style of using java code as configuration.
Maybe have a `JkRepoAuthentication` interface that intercepts the HTTP request before sending to add information such as header or even query parameter. The `JkRepoCredentials` class could then implement said interface with the current logic. 
But just as an idea.

Thank you for your quick response.

**Comment by djeang** [comment]
I think something simpler as JkRepo#setHttpHeaders(Map<String, String>).
The headers would be directly set on the repo object so there would be no obscure mechanism to deal with.

**Comment by rheinertleon** [comment]
This definintely is easier and even supports other cases, where one might want to add headers 👍

**Comment by djeang** [comment]
The feature is now available in the last version (0.10.5). Reopen the issue if you encounter any problem.

By the way, I am willing to get feedback on JeKa : where you  are using it (side projects, work, os projects). What do you feel about it (nice, missing feature,..), does anyone know about it around you ?

Cheers

**Comment by rheinertleon** [comment]
Works like a charm, thanks again for your quick help.

I found out about Jeka for this work-related project, where I need to download maven dependencies at runtime.
But I genuinely like the actual purpose of a build tool, which I will try out for my next side project (if I find the time tho).

So far, I'm a fan of the easy to use API, but will let you know when I use Jeka in more detail. 😄
